### DISCUSSION TITLE
Struct Overlapping Single Variables on Stack

### DISCUSSION THREAD

**Discussion by utkonos** [Vector35/binaryninja-api#5260] [Q&A] [answered]
When a particular address on the stack is used for different variables, they're all created automatically and nicely (this is just an examples of overlap at an address; these particular variables are totally not related to the problem below):
<img width="300" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/2775739/fb696eb0-f464-4a3d-b9ab-5a1304437bfc">

However, if I want to set the type to a struct that overlaps the locations used previously on the stack, after setting that type, the other varables are gone. Is there a way to get this to work?

Here is how the Stack view looks before. That struct is incorrectly a pointer at the moment just as a placeholder.
<img width="283" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/2775739/5cbc8c4d-ba5d-4329-ad78-d3ae1b362753">

The two variables enc_str.size and fill_value are both input parameters for init_string here:
<img width="585" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/2775739/223dd1ba-170a-4617-8c9e-ff1a083b5a65">

That function cleans up the stack via `ret 8`.

Later in the function, space is created for the struct on the stack via `sub esp, 0x18`. For some reason, if I set the type at that location to the struct, it blows away those two previous stack variables.
<img width="298" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/2775739/d45d390e-f61e-4fe2-9abe-607635bdfe92">

I want to keep those two variables in the second screenshot above and set a struct at the same addresses in the stack. This way it reflects both usages at the different points in the function.

Is what I want to do possible? I can always just write comments to work around this, but the downstream uses of that struct don't show up nicely with the members in the disassembly view.

**Comment by fuzyll** [comment, marked as answer]
I do not believe what you want to do is currently possible. But, I think we *should* make it possible. I'm going to convert this to a feature request as a result.
